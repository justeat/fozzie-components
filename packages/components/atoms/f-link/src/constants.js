export const VALID_LINK_TYPES = ['default', 'external', 'newLocation'];
export const DEFAULT_LINK_TYPE = 'default';
