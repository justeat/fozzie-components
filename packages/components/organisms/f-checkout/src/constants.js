export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHODS = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];

export const TENANT_MAP = {
    'en-GB': 'uk',
    'en-AU': 'au',
    'en-NZ': 'nz',
    'da-DK': 'dk',
    'es-ES': 'es',
    'en-IE': 'ie',
    'it-IT': 'it',
    'nb-NO': 'no'
};

export const BASKET_NOT_ORDERABLE_ERRORS = ['RESTAURANT_NOT_TAKING_ORDERS', 'SERVICE_TYPE_UNAVAILABLE', 'ITEMS_UNORDERABLE', 'ADDITIONAL_ITEMS_REQUIRED', 'MINIMUM_ORDER_VALUE_NOT_MET', 'LOCATION_UNDELIVERABLE', 'AGE_VERIFICATION_FAILED'];
export const INVALID_MODEL_STATE_ERRORS = ['FIRST_NAME_REQUIRED', 'LAST_NAME_REQUIRED', 'PHONE_NUMBER_REQUIRED', 'DATE_OF_BIRTH_REQUIRED', 'FULFILMENT_TIME_REQUIRED', 'ADDRESS_LINES_REQUIRED', 'POSTAL_CODE_REQUIRED', 'GEO_LOCATION_REQUIRED'];
export const SET_ORDER_TIME_ERROR = 'FULFILMENT_TIME_UNAVAILABLE'

export const VALIDATIONS = {
    address: 'addressValidations',
    guest: 'guestValidations'
};

export const VUEX_CHECKOUT_MODULE = 'fCheckoutModule';
export const VUEX_CHECKOUT_ANALYTICS_MODULE = 'fCheckoutAnalyticsModule';
