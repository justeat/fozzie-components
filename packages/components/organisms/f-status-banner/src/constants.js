export const BASE_HERO_URL = 'https://d30v2pzvrfyzpo.cloudfront.net/status-banner/';
