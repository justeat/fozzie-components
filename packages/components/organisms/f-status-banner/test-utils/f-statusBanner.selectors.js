export const COMPONENT = '[data-test-id="main-banner-container"]';
export const ADDRESS_BOX = '[data-test-id="address-box-input"]';
export const BUTTON = '[data-test-id="find-restaurants-button"]';
