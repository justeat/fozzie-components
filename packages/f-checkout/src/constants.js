export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHOD = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];
