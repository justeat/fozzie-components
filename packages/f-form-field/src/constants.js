export const VALID_INPUT_TYPES = ['text', 'email', 'password', 'radio', 'checkbox'];
export const CUSTOM_INPUT_TYPES = ['dropdown'];
export const DEFAULT_INPUT_TYPE = 'text';

export const VALID_LABEL_STYLES = ['default', 'inline', 'inlineNarrow'];

export const VALID_GROUP_POSITION = ['top', 'bottom', 'middle'];

export const MOBILE_WIDTH = 768;
