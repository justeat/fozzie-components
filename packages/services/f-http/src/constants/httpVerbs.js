const METHOD_GET = 'GET';
const METHOD_POST = 'POST';

export default {
    METHOD_GET,
    METHOD_POST
};
