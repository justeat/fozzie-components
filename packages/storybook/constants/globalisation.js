export const AUSTRALIAN_LOCALE = 'en-AU';
export const DANISH_LOCALE = 'da-DK';
export const ENGLISH_LOCALE = 'en-GB';
export const IRISH_LOCALE = 'en-IE';
export const ITALIAN_LOCALE = 'it-IT';
export const NEW_ZEALAND_LOCALE = 'en-NZ';
export const NORWEIGAN_LOCALE = 'nb-NO';
export const SPANISH_LOCALE = 'es-ES';

export const VALID_LOCALES = [ENGLISH_LOCALE, SPANISH_LOCALE];
