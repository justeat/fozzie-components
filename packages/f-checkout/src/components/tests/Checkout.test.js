import { shallowMount } from '@vue/test-utils';
import { VALID_CHECKOUT_TYPES } from '../../constants';
import VueCheckout from '../Checkout.vue';

describe('Checkout', () => {
    it('should be defined', () => {
        const propsData = {};
        const wrapper = shallowMount(VueCheckout, { propsData });
        expect(wrapper.exists()).toBe(true);
    });

    describe('props ::', () => {
        describe('checkoutType ::', () => {
            it.each(VALID_CHECKOUT_TYPES)('should set load the component as expected if checkoutType=%p is specified', definedType => {
                // Arrange
                const propsData = {
                    checkoutType: definedType
                };

                // Act
                const wrapper = shallowMount(VueCheckout, { propsData });
                const selectedComponent = wrapper.find('[data-test-id="checkout-component-type"]');

                // Assert
                expect(selectedComponent.element.tagName).toMatchSnapshot();
            });
        });
    });

    describe('computed ::', () => {
        const propsData = {};
        const data = { firstName: 'name' };

        describe('name ::', () => {
            it('should capitalize `firstName` data', async () => {
                // Arrange
                const wrapper = shallowMount(VueCheckout, { propsData });
                const name = wrapper.find("[data-test-id='checkout-card-component']");

                // Act
                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                // Assert
                expect(name.props('cardHeading')).toContain('Name');
            });
        });

        describe('title ::', () => {
            it('should add `name` to title text', async () => {
                // Arrange
                const wrapper = shallowMount(VueCheckout, { propsData });
                const name = wrapper.find("[data-test-id='checkout-card-component']");

                // Act
                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                // Assert
                expect(name.props('cardHeading')).toEqual('Name, confirm your details');
            });
        });
    });
});
