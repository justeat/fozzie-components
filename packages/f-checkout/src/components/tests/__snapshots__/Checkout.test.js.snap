// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout props :: checkoutType :: should set load the component as expected if checkoutType="Collection" is specified 1`] = `"COLLECTION-STUB"`;

exports[`Checkout props :: checkoutType :: should set load the component as expected if checkoutType="Delivery" is specified 1`] = `"DELIVERY-STUB"`;
